packages/gestalt/src/SideNavigation/ItemContent.tsx

```diff
@@ -0,0 +1,258 @@
+          dangerouslySetInlineStyle={{ __style: { top: 4, right: 4 } }}
+          height={8}
+          position="absolute"
+          role="status"
+          rounding="circle"
+          width={8}
+        />
+      ) : null}
+
+      <Flex
+        alignItems="center"
+        flex="grow"
+        gap={{ row: 2, column: 0 }}
+        height="100%"
+        justifyContent={collapsed ? 'center' : undefined}
+      >
+        {icon ? (
+          <Flex.Item>
+            <Box aria-hidden={!collapsed}>
+              {typeof icon === 'string' ? (
+                <Icon
+                  accessibilityLabel={collapsed ? label : ''}
+                  color={textColor}
+                  icon={icon}
+                  inline
+                  size={20}
+                />
+              ) : (
+                <Icon
+                  accessibilityLabel={collapsed ? label : ''}
+                  color={textColor}
+                  dangerouslySetSvgPath={icon}
+                  inline
+                  size={20}
+                />
+              )}
+            </Box>
+          </Flex.Item>
+        ) : null}
+
+        {!collapsed && (
+          <Flex.Item flex="grow">
+            <Text color={textColor} inline>
+              {label}
+              {(badge || notificationAccessibilityLabel) && (
+                <Box display="inlineBlock" height="100%" marginStart={1}>
+                  {/* Adds a pause for screen reader users between the text content */}
+                  <Box display="visuallyHidden">{`, `}</Box>
+                  {!notificationAccessibilityLabel && badge ? (
+                    <Badge text={badge.text} type={badge.type} />
+                  ) : null}
+                  {notificationAccessibilityLabel ? (
+                    <Box
+                      aria-label={notificationAccessibilityLabel}
+                      color="primary"
+                      height={8}
+                      role="status"
+                      rounding="circle"
+                      width={8}
+                    />
+                  ) : null}
+                </Box>
+              )}
+            </Text>
+          </Flex.Item>
+        )}
+
+        {!collapsed && counter ? (
+          <Flex.Item flex="none">
+            <Box display="visuallyHidden">{`, `}</Box>
+            <Box aria-label={counter.accessibilityLabel}>
+              <Text align="end" color={counterColor}>
+                {counter.number}
+              </Text>
+            </Box>
+          </Flex.Item>
+        ) : null}
+
+        {!collapsed && primaryAction ? (
+          <Flex.Item flex="none">
+            {/* This is a workaround to announce the counter as it's replaced on focus */}
+            {counter ? (
+              <Box display="visuallyHidden">
+                {`, `}
+                <Box aria-label={counter?.accessibilityLabel} />
+              </Box>
+            ) : null}
+            <Box aria-hidden marginEnd={-2} rounding="circle">
```

Comment:
@AlbertCarreras @hectoid I don't understand why PrimaryActionButton should be "aria-hidden". Anything I am missing here?

Reply:
I did this. 

The IconButton still gets announced when you focus on it on keyboard navigation. 

aria-hidden prevents this Button to be read as part of the parent Button. Otherwise all the Buttons would be read in the parent one. You want to hide that Button on Buttons content, so it only announces the parent one, and the the children when focused. Makes sense?

Reply:
should we add comment to the code?